src/serde.rs

```diff
@@ -7,12 +7,37 @@
 /// directly to structs this is th4 place to go.
 mod de;
 mod value;
+pub use self::value::*;
 use crate::numberparse::Number;
 use crate::{stry, Deserializer, Error, ErrorType, Result};
+use crate::{BorrowedValue, OwnedValue};
 use serde_ext::Deserialize;
+use std::convert::{TryFrom, TryInto};
 use std::fmt;
 
-pub use self::value::*;
+type ConvertResult<T> = std::result::Result<T, SerdeConversionError>;
```

Comment:
this makes sense 👍 

is it feasible to add test(s) to make sure patterns are covered?

Reply:
Absolutely! It's not done yet going to add serde_yaml too, sorry I should have marked it as a draft.